src/Components/Todo/TodoListItem.tsx

```diff
@@ -0,0 +1,166 @@
+/* eslint-disable jsx-a11y/label-has-associated-control */
+import { useContext } from 'react';
+import { Todo } from '../../types/Todo';
+import { TodoContext } from '../TodoContext/TodoContext';
+import classNames from 'classnames';
+import { useState } from 'react';
+import { deleteTodo, updateTodo } from '../../api/todos';
+
+type Props = {
+  todo: Todo;
+};
+
+export const TodoListItem: React.FC<Props> = ({ todo }) => {
+  const { todosList, setTodosList, setErrorMessage } = useContext(TodoContext);
+
+  const { id, title, completed } = todo;
+
+  const [editable, setEditable] = useState(false);
+
+  const [newTitle, setNewTitle] = useState(title);
+
+  const [trimmedTitle, setTrimmedTitle] = useState(newTitle.trim());
```

Comment:
redundant state - just trim it where you need trimmed value (save in common `const` variable)